clients/python/test/e2e/ci/_data_classes.py

```diff
@@ -0,0 +1,172 @@
+import configparser
+from pathlib import Path
+from typing import Dict, Optional
+from urllib.parse import ParseResult, urlparse
+
+from _utils import _PYTEST_INI
+from packaging.version import Version
+from pydantic import BaseModel, field_validator, model_validator
+
+# Holds classes for passing data around between scripts.
+
+
+class ServerConfig(BaseModel):
+    """Holds data about server configuration"""
+
+    osparc_api_host: str
+    osparc_api_key: str
+    osparc_api_secret: str
+
+    @property
+    def url(self) -> ParseResult:
+        return urlparse(f"{self.osparc_api_host}")
+
+    @property
+    def key(self) -> str:
+        return self.osparc_api_key
+
+    @property
+    def secret(self) -> str:
+        return self.osparc_api_secret
+
+
+def is_empty(v):
+    return v is None or v == ""
+
+
+class ClientConfig(BaseModel):
+    """Holds data about client configuration.
+    This data should uniquely determine how to install client
+    """
+
+    osparc_client_version: Optional[str] = None
+    osparc_client_repo: Optional[str] = None
+    osparc_client_branch: Optional[str] = None
+    osparc_client_workflow: Optional[str] = None
+    osparc_client_runid: Optional[str] = None
+
```

Comment:
just wondering about the naming. since you always create properties with shorter ones.
Why not calling the class ```OsparcClientConfig``` and then have the fields as ```version, repo, branch, workflow, runid```?

Reply:
Good point. The reason for this is that all the configurations are carried around as environment variables in the gitlab ci scripts. There a single environment variable is an array of json objects, each of which is a server configuration. Hence, in those yml files it is nicer to have more meaningful names. I then have the getters with shorter names in the python classes... I don't know if this is the perfect approach but it is a compromise to have readable yml files on the gitlab repo and easy python code here...